app/Knights/knight.py

```diff
@@ -0,0 +1,32 @@
+from app.Staff.armour import Armour
+from app.Staff.potion import Potion
+from app.Staff.weapon import Weapon
+
+
+class Knight:
+    knights = {}
+
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            health_points: int,
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.health_points = health_points
+        self.protection = 0
+        Knight.knights = {self.name: self}

```

Comment:
The assignment `Knight.knights = {self.name: self}` will overwrite the `knights` dictionary every time a new `Knight` instance is created. If the intention is to keep track of all knights, consider using `Knight.knights[self.name] = self` instead.